htdocs/projet/class/task.class.php

```diff
@@ -1505,6 +1505,14 @@ public function addTimeSpent($user, $notrigger = 0)
 				$this->error = $this->db->lasterror();
 				$ret = -2;
 			}
+
+			// update note in table
+			$nsql = "UPDATE ".MAIN_DB_PREFIX."element_time SET note = '".$this->db->escape($timespent->note)."'";
```

Comment:
This looks strange because saving the $timespent->note should be already done on line 1501 with
$timespent->update()
What sql do yo have when the timespent->update is done ?

Reply:
the problem is the commonobject update function used to initialize the other columns to null, which causes the value of the “note” field to be lost.

Reply:
Line 1510 is inside $timespent->create().
It used createCommon()
But before calling ->create(), we have
$timespent->note = $this->timespent_note;
So the createCommon should create record with note = value defined into line
$timespent->note = $this->timespent_note;

But the update should also do update of the note from the value 
$timespent->note

So the problem is surely why the note is set to null. it should not because the property $timespent->note should be set with value to use for the update.